### ISSUE TITLE
Storage Account doesn't get created.

### ISSUE THREAD

**Issue by kishan-polekar-cof** [Azure/azure-sdk-for-go#23886]
I'm trying to follow the example code [here](https://learn.microsoft.com/en-us/rest/api/storagerp/storage-accounts/create?view=rest-storagerp-2023-05-01&tabs=Go#storageaccountcreate). I have my handler set up so that a 200 response code can be received, but I get a 201 each time. And when I try to check my Azure portal, it doesn't have the storage account within an already existing resource group. There is no proper error or response when I try to output it on the console. This is what I get as a response for `clientFactory.NewAccountsClient().BeginCreate` as well as for `poller.PollUntilDone(ctx, nil)` (both have almost same content): 

`\u003cnil\u003e \u003cnil\u003e \u003cnil\u003e \u003cnil\u003e map[] \u003cnil\u003e \u003cnil\u003e \u003cnil\u003e \u003cnil\u003e \u003cnil\u003e`

There are no further errors, and even after trying to run the sample code, I don't see a proper response (without changing any parameters).

**Comment by jliusan** [comment]
@kishan-polekar-cof  could you turn on logging by setting environment variable AZURE_SDK_GO_LOGGING to all and configure policy.LogOption to print the body of this call? Please help to check the details info in the raw response.

https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/new-version-guideline.md#logging

**Comment by kishan-polekar-cof** [comment]
Hi @jliusan I was able to figure out the issue without the need of logging as you mentioned. This will definitely help in the future if I have any problems, but seems like the issue I had was due to incorrect variables being set in the `NetworkRuleSet` parameter (specifically the `VirtualNetworkResourceID` which was taking empty string values during formatting instead of the actual values set globally. Once I fixed that, my storage account was created. There were also some policies blocking the creation but setting the correct parameters helped bypass those as well. Thanks for your help, this issue can be closed!
